birdhouse/config/jupyterhub/config/magpie/docker-compose-extra.yml

```diff
@@ -0,0 +1,5 @@
+version: "3.4"
+services:
+  magpie:
+    environment:
+      MAGPIE_USER_NAME_EXTRA_REGEX: "^[a-z0-9]+$$"
```

Comment:
Just curious, instead of this env var could we have done this in the `magpie.ini` file?  I have a slight preference for all configs to be grouped together in one file for ease of tracing them.

Is the `magpie.ini` route not possible or there are advantages to use env var?

Reply:
The reason that I put this here is that we only need to enforce this limitation if both the magpie and jupyterhub components are enabled. 
Yes, this will be 99% of the use-cases but it's important to keep the dependencies clear in the way we structure things in this repo.

Reply:
Another way to think about this: This follows the "pluggable component architecture" that we've been trying to create since #214

Reply:
Oh I see !  Never thought about that one, having Magpie without JupyterHub do not need this config.  You are right.

Reply:
Hey, I just think of a counter argument in favor of `magpie.ini` since we are discussing about control users already violating the extra regex.

Even if Magpie can initially be activated without Jupyterhub, we better ensure all the users created are "JupyterHub forward compatible".  Otherwise the poor node admin will be in for a surprise the day they activate JupyterHub.

That is to say, it's more future proof to have a stricter regex, then relax it if needed, than the other way around.

Reply:
@mishaschwartz Not sure if you've seen my previous comment above but I think it's better to have the new extra regex in `magpie.ini` so all users created are forward-compatible if JupyterHub is enabled later.